logstash-core/lib/logstash/runner.rb

```diff
@@ -601,4 +604,16 @@ def setting(key)
     @settings.get_value(key)
   end
 
+  def configure_io_receive_buffer_allocation_to_heap
+    # if user hasn't explicitly set Netty's interested properties
+    if java.lang.System.getProperty("io.netty.allocator.numDirectArenas") == nil &&
+        java.lang.System.getProperty("io.netty.noPreferDirect")  == nil
+      # set Netty's Java properties to disable direct memory allocation
+      java.lang.System.setProperty("io.netty.allocator.numDirectArenas", "0")
```

Comment:
Given that we can't prevent direct memory allocations, I don't think we should set the arenas to 0. Without arenas it just means that any direct memory allocation will be unpooled and thus slower than when arenas are used. I'd suggest either leaving the default or setting to 1.